opts.go

```diff
@@ -66,3 +69,17 @@ func WithSnapshot(memFilePath, snapshotPath string, opts ...WithSnapshotOpt) Opt
 		m.Handlers.FcInit = loadSnapshotHandlerList
 	}
 }
+
+// MemoryBackendType sets the memory backend type to the given value.
+func MemoryBackendType(typ string) WithSnapshotOpt {
```

Comment:
Since firecracker [deprecated](https://github.com/firecracker-microvm/firecracker/blob/main/src/api_server/swagger/firecracker.yaml#L1196) `memFilePath `, I think we should use the later approach in case it got removed in the future.

Reply:
> Since firecracker [deprecated](https://github.com/firecracker-microvm/firecracker/blob/main/src/api_server/swagger/firecracker.yaml#L1196) memFilePath , I think we should use the later approach in case it got removed in the future.

Do you mean that we should stop setting `MemFilePath`, and only set `MemBackend`? If yes, I think this will break users who are on older versions of firecracker that don't support `MemBackend`. Is that OK?

Also, I should have mentioned this earlier, but the reason that my initial implementation of `WithMemoryBackend` was so complex because it was trying to be backwards compatible with older firecracker versions. It would always use the deprecated `MemFilePath` (for backwards compatibility), unless `WithMemoryBackend` was used, which would "move" the `MemFilePath` into `MemBackend.BackendPath`. This was the code: https://github.com/firecracker-microvm/firecracker-go-sdk/blob/62f8e172aec24033fc99c007ef093d4f10008e45/opts.go#L74-L85

Reply:
> Do you mean that we should stop setting MemFilePath, and only set MemBackend?

I was more leaning to who started using `MemBackend` which is only available in latest firecracker version should use 
```
WithSnapshot(
  "" /*memFile*/, snapshotPath,
  WithMemoryBackend("UFFD", "/path/to/uffd.sock"))
```
instead of 
```
WithSnapshot(
  "/path/to/uffd.sock", snapshotPath,
  WithMemoryBackend("UFFD"))
```
Those who are still using an old firecracker version(assuming they don't use `MemBackend`), they can still do, don't they? 
```
WithSnapshot(
  "" /*memFile*/, snapshotPath)
```
But since firecracker changed `memFile` from a `* string` to `string`, users want to move to new firecracker-go-sdk version anyway need to update their code(backwards compatibility broke).

Reply:
OK. If I understand correctly, you are saying we should keep the current code as-is. If not, please let me know.

Reply:
Yeah, keep the current code as it is looks good to me.